File: QViz.NET/Periscope.QViz/JSON/Permission.cs
using System;
using System.Collections.Generic;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for User Permissions
	/// </summary>
	public class Permission
	{
		/// <summary>
		/// Serial Number of the User Permission Object
		/// </summary>
		public int srno;
		
		/// <summary>
		/// Identifier of the User Permission Object
		/// </summary>
		public string permissionId;
		
		/// <summary>
		/// Name of the User Permission Object
		/// </summary>
		public string name;
		
		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;
		
		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;
		
		/// <summary>
		/// Is this User Permission Active?
		/// </summary>
		public bool isActive;
		
		/// <summary>
		/// List of User Role Permission Objects
		/// </summary>
		public IList<RolePermission> rolePermissions;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public Permission()
		{
			this.srno = 0;
			this.permissionId = "";
			this.name = "";
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
			this.isActive = false;
			this.rolePermissions = new List<RolePermission>();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/Permission.java
package com.periscope.qviz.json;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * JSON Model for User Permissions
 */
public class Permission {

	/**
	 * Serial Number of the User Permission Object
	 */
	public Integer srno;

	/**
	 * Identifier of the User Permission Object
	 */
	public String permissionId;

	/**
	 * Name of the User Permission Object
	 */
	public String name;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	/**
	 * Is this User Permission Active?
	 */
	public Boolean isActive;

	/**
	 * List of User Role Permission Objects
	 */
	public List<RolePermission> rolePermissions;

	/**
	 * Default Constructor
	 */
	public Permission() {
		this.srno = 0;
		this.permissionId = "";
		this.name = "";
		this.createdOn = null;
		this.updatedOn = null;
		this.isActive = false;
		this.rolePermissions = new ArrayList<>();
	}

}
